### PR TITLE
Adds alpine based image

### PR DESCRIPTION
**Pull request by kenjones-cisco**
Adds Dockerfile and docker-entrypoint.sh under version specific alpine
directory.
Update BUILDING to demonstrate the inclusion of a tagged alpine version.

### CODE REVIEW

File: 4.0.0/alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Here we install GNU libc (aka glibc) and set C.UTF-8 locale as default.
+# required to use 'rqlited' within alpine as it is not statically linked
+RUN ALPINE_GLIBC_BASE_URL="https://github.com/sgerrand/alpine-pkg-glibc/releases/download" && \
+    ALPINE_GLIBC_PACKAGE_VERSION="2.23-r3" && \
+    ALPINE_GLIBC_BASE_PACKAGE_FILENAME="glibc-$ALPINE_GLIBC_PACKAGE_VERSION.apk" && \
+    ALPINE_GLIBC_BIN_PACKAGE_FILENAME="glibc-bin-$ALPINE_GLIBC_PACKAGE_VERSION.apk" && \
+    ALPINE_GLIBC_I18N_PACKAGE_FILENAME="glibc-i18n-$ALPINE_GLIBC_PACKAGE_VERSION.apk" && \
+    apk add --no-cache --virtual=.build-dependencies wget ca-certificates && \
+    wget \
+        "https://raw.githubusercontent.com/andyshinn/alpine-pkg-glibc/master/sgerrand.rsa.pub" \
```
Comment (otoolep) [inline 4.0.0/alpine/Dockerfile:13]: What is the source of this cert, key, library, etc? I don't want to publish Docker images unless I understand the source of the glibc inside.

**Comment by otoolep** [comment]
Thanks @kenjones-cisco -- one question about origin.

File: 4.0.0/alpine/Dockerfile (hunk 1 above)
Comment (kenjones-cisco) [reply 4.0.0/alpine/Dockerfile:13]: The glibc is built using this project:
https://github.com/sgerrand/docker-glibc-builder

The source for glibc as indicated by the file `builder` within the project is:
http://ftp.gnu.org/gnu/glibc/glibc-$version.tar.gz

The pub key passed in allows Alpine `apk` to verify the package that is being installed. The private key is used in the actual build process. That is related to the internals of the how the `apk` package manager works.

Comment (otoolep) [reply 4.0.0/alpine/Dockerfile:13]: Got it -- thanks. I will need to study this process before pushing out rqlite packages with libraries built elsewhere.

**Comment by otoolep** [comment]
Thanks for your contribution, but we've actually got an Alpine image now.

**Pull request closed without merging**